tests/rkt_mount_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/coreos/rkt/tests/testutils"
+)
+
+func TestMountSymlink(t *testing.T) {
+	mountSrcFile := filepath.Join(os.TempDir(), "hello")
+	if err := ioutil.WriteFile(mountSrcFile, []byte("world"), 0666); err != nil {
+		t.Fatalf("Cannot write file: %v", err)
+	}
+	defer os.Remove(mountSrcFile)
+
+	image := patchTestACI("rkt-test-mount-symlink.aci", "--exec=/inspect --read-file")
+	defer os.Remove(image)
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	tests := []struct {
+		linkFile     string
+		actualFile   string
+		expectedLine string
+		exitCode     int
+	}{
+		// '/dir1/link_abs_dir2' links to '/dir2'.
+		{
+			"/dir1/link_abs_dir2/foo",
+			"/dir2/foo",
+			"world",
+			0,
+		},
+		// '/dir1/link_rel_dir2' links to '/dir2'.
+		{
+			"/dir1/link_rel_dir2/bar",
+			"/dir2/bar",
+			"world",
+			0,
+		},
+		// '/dir1/../dir1/.//link_rel_dir2' links to '/dir2'.
+		{
+			"/dir1/../dir1/.//link_rel_dir2/foo",
+			"/dir2/foo",
+			"world",
+			0,
+		},
+		// '/dir1/../../../foo' is invalid because it tries to escape rootfs.
+		{
+			"/dir1/../../../foo",
+			"/dir2/foo",
+			"path escapes app's root",
+			3,
```

Comment:
If the code is changed as suggested above, this test case will not error out anymore but resolve to `/foo`.